srepanel/web/request_access.go

```diff
@@ -0,0 +1,114 @@
+package web
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"go.mkw.re/ghidra-panel/common"
+)
+
+// https://discord.com/developers/docs/resources/channel#embed-object-embed-author-structure
+type AccessRequestEmbedAuthor struct {
+	Name    string `json:"name"`
+	IconURL string `json:"icon_url"`
+}
+
+// https://discord.com/developers/docs/resources/channel#embed-object-embed-field-structure
+type AccessRequestEmbedField struct {
+	Name   string `json:"name"`
+	Value  string `json:"value"`
+	Inline bool   `json:"inline"`
+}
+
+// https://discord.com/developers/docs/resources/channel#embed-object
+type AccessRequestEmbed struct {
+	Title  string                    `json:"title"`
+	Color  int                       `json:"color"`
+	Author AccessRequestEmbedAuthor  `json:"author"`
+	Fields []AccessRequestEmbedField `json:"fields"`
+}
+
+// https://discord.com/developers/docs/resources/webhook#execute-webhook
+type AccessRequestMessage struct {
+	Username  string               `json:"username"`
+	AvatarURL string               `json:"avatar_url"`
+	Embeds    []AccessRequestEmbed `json:"embeds"`
+}
```

Comment:
These should be moved into a `discord` package and named after the API types, so we can reference them with e.g. `discord.EmbedAuthor`. Alternatively we can import https://github.com/bwmarrin/discordgo

Reply:
Sounds good to me

Reply:
Resolved in 22ccd92.